lib/request-handler.js

```diff
@@ -0,0 +1,148 @@
+"use strict";
+
+
+/**
+ * @typedef {Object} requestHandler
+ * @function buildRequestOptions
+ * @function checkResponse
+ * @function createUrl
+ */
+
+
+/**
+ * Creates requestHandler.
+ *
+ * @param {bluebird} bluebird
+ * @param {dateService} dateService
+ * @param {string} host
+ * @param {logger} logger
+ * @param {urijs} Uri
+ * @return {requestHandler}
+ */
+module.exports = (bluebird, dateService, host, logger, Uri) => {
+    // This error object is a last ditch attempt to supply info if
+    // OpenToken does not provide a body.
+    const error = {
+        401: "unauthorized",
+        403: "forbidden",
+        404: "resource_not_found",
+        405: "method_not_allowed",
+        500: "internal_server_error",
+        unknown: "unknown"
+    };
+
+
+    /**
+     * Build request options for a request to OpenToken.
+     *
+     * @param {string} url
+     * @param {string} method
+     * @param {string} contentType
+     * @return {Object}
+     */
+    function buildRequestOptions(url, method, contentType) {
+        if (typeof contentType === "undefined") {
```

Comment:
Do you care about only `undefined`?  I think that `!contentType` is a better check in this instance.